src/node/txreconciliation.cpp

```diff
@@ -115,10 +133,66 @@ class TxReconciliationTracker::Impl
                       peer_id, is_peer_inbound);
 
         const uint256 full_salt{ComputeSalt(local_salt, remote_salt)};
-        recon_state->second = TxReconciliationState(!is_peer_inbound, full_salt.GetUint64(0), full_salt.GetUint64(1));
+
+        auto new_state = TxReconciliationState(!is_peer_inbound, full_salt.GetUint64(0), full_salt.GetUint64(1));;
+        m_states.erase(recon_state);
+        m_states.emplace(peer_id, std::move(new_state));
+
         return ReconciliationRegisterResult::SUCCESS;
     }
 
+    bool AddToSet(NodeId peer_id, const Wtxid& wtxid) EXCLUSIVE_LOCKS_REQUIRED(!m_txreconciliation_mutex)
+    {
+        AssertLockNotHeld(m_txreconciliation_mutex);
+        LOCK(m_txreconciliation_mutex);
+        auto peer_state = GetRegisteredPeerState(peer_id);
+        if (!peer_state) return false;
+
+        // Check if the reconciliation set is not at capacity for two reasons:
+        // - limit sizes of reconciliation sets and short id mappings;
+        // - limit CPU use for sketch computations.
+        //
+        // Since we reconcile frequently, reaching capacity either means:
+        // (1) a peer for some reason does not request reconciliations from us for a long while, or
+        // (2) really a lot of valid fee-paying transactions were dumped on us at once.
+        // We don't care about a laggy peer (1) because we probably can't help them even if we fanout transactions.
+        // However, exploiting (2) should not prevent us from relaying certain transactions.
+        //
+        // Transactions which don't make it to the set due to the limit are announced via fan-out.
+        if (peer_state->m_local_set.size() >= MAX_RECONSET_SIZE) return false;
+
+        // The caller currently keeps track of the per-peer transaction announcements, so it
+        // should not attempt to add same tx to the set twice. However, if that happens, we will
+        // simply ignore it.
+        if (peer_state->m_local_set.insert(wtxid).second) {
+            LogPrintLevel(BCLog::TXRECONCILIATION, BCLog::Level::Debug, "Added %s to the reconciliation set for peer=%d. " /* Continued */
+                                                                        "Now the set contains %i transactions.\n",
+                          wtxid.ToString(), peer_id, peer_state->m_local_set.size());
+        }
+        return true;
+    }
+
+    bool TryRemovingFromSet(NodeId peer_id, const Wtxid& wtxid) EXCLUSIVE_LOCKS_REQUIRED(!m_txreconciliation_mutex)
+    {
+        AssertLockNotHeld(m_txreconciliation_mutex);
+        LOCK(m_txreconciliation_mutex);
+        auto peer_state = GetRegisteredPeerState(peer_id);
+        if (!peer_state) return false;
+
+        auto removed = peer_state->m_local_set.erase(wtxid) > 0;
+        if (removed) {
+            LogPrintLevel(BCLog::TXRECONCILIATION, BCLog::Level::Debug, "Removed %s from the reconciliation set for peer=%d. " /* Continued */
```

Comment:
https://github.com/bitcoin/bitcoin/commit/d2f260507279b1edd6c0f55d02c7ae2e598b3585

nit: in three places in this commit `/* Continued */` should be dropped, it's no longer necessary (been a linter thing)